hooks/command

```diff
@@ -182,6 +182,9 @@ status=""
 if [[ "$jobstatus" == "Complete" ]] ; then
   echo "success"
   status="0"
+elif kubectl get pod "$pod_name" 2>&1 >/dev/null | grep -q NotFound; then
+  echo "Warning: pod was never created"
+  status="42"
```

Comment:
This makes sense! if the pod doesn't exist for whatever reason then no need to retry. Could we move the check into the else statement  rather than a separate block? it'll allow us to also skip the whole init checking logic if we don't have a pod spec to begin with: e.g

```bash
...
pod_json=$(kubectl get pod "$pod_name" -o json 2>&1)
# Only check iniit container if we have a pod spec
if [[ get_pod_status succeeded ]]; then
  init_container_status="$(echo "$pod_json" | jq ".status.initContainerStatuses[0].state.terminated.exitCode")"
  if [[ -n "$init_container_status" && "$init_container_status" != "0" ]]; then
      echo "Warning: init container failed with exit code $init_container_status, this usually indicates plugin misconfiguration or infrastructure failure"
      status="$init_container_status"
  else
    status="$(echo "$pod_json" | jq ".status.containerStatuses[0].state.terminated.exitCode")"
  fi
else if [[ $pod_json contains NotFound ]]; then
  echo "Warning:..."
  status=42
else
  echo $pod_json # contains command's stderr
fi
...
```

Reply:
Are you suggesting we drop the while loop that's below this entirely or were you omitting to be concise?

Reply:
I've updated the PR with this approach for discussion, one thing I noticed while doing it is that the JSON might match NotFound elsewhere and give us a false positive? Not sure how realistic that is. Also if kubectl outputs _anything_ to stderr it will likely produce invalid json since we're combining these here

Reply:
> Are you suggesting we drop the while loop that's below this entirely or were you omitting to be concise?

Yeah we still need the loop as is, it was to be concise.

> I've updated the PR with this approach for discussion, one thing I noticed while doing it is that the JSON might match NotFound elsewhere and give us a false positive? Not sure how realistic that is. Also if kubectl outputs anything to stderr it will likely produce invalid json since we're combining these here (not sure if it does that for warnings or anything)

I think both issues are handled correctly by checking for an error in the output only if the command failed which we seem to be doing  - we only get a json from kubectl if the command succeeded, if we get anything in stderr the command failed to, in that case it should be okay to treat any 'NotFound' as pod not found error (technically the error might be that something else wasn't found but that's unlikely and can be resolved later on if needed)

Also sorry for the delayed response